packages/owl-bot/src/copy-code.ts

```diff
@@ -329,187 +249,249 @@ export function branchNameForCopy(yamlPath: string): string {
+
+      if (pull) {
+        // Yes, there's a pull request.  Append a new commit to it.
+        appendedPullCount += 1;
+        logger.info(`Appending ${pull.html_url} with a new commit.`);
+        cmd(`git checkout -t origin/${destBranch}`, {cwd: destDir});
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const reportError = async (e: any, yamlPath: string) => {
+          console.error(`Error parsing ${yamlPath}: ${e}`);
+          await octokit.pulls.createReviewComment({
+            repo: destRepo.repo,
+            owner: destRepo.repo,
+            pull_number: pull.number,
+            body: `Error parsing ${yamlPath}.  Triggered by ${sourceLinkFrom(
+              sourceRepoCommitHash
+            )}:\n${e}`,
+          });
+          const copyTag = copyTagFrom(yamlPath, sourceRepoCommitHash);
+          await copyStateStore.recordBuildForCopy(
+            destRepo,
+            copyTag,
+            pull.html_url
+          );
+        };
+
+        // Copy the code from googlapis-gen.
+        const copiedYamls = await copyCodeIntoCommit(
+          sourceRepo,
+          sourceRepoCommitHash,
+          destDir,
+          workDir,
+          yamlPathSet,
+          reportError,
+          logger
+        );
+
+        if (copiedYamls.length === 0) {
+          continue; // No changes to push.
+        }
+
+        // Push the changes to the pull request.
+        const token = await octokitFactory.getGitHubShortLivedAccessToken();
+        const pushUrl = destRepo.getCloneUrl(token);
+        cmd(`git push ${pushUrl} ${destBranch}`, {cwd: destDir});
+
+        // Record that we've copied the code.
+        for (const yaml of copiedYamls) {
+          await copyStateStore.recordBuildForCopy(
+            destRepo,
+            yaml.copyTag,
+            pull.html_url
+          );
+        }
+
+        // Prepend the new commit message to the body.
+        const commitBody: string = cmd('git log -1 --format=%B', {
+          cwd: destDir,
+        })
+          .toString('utf8')
+          .trim();
+        const {title, body} = resplit(
+          `${commitBody}\n\n` +
+            `${pull.title}\n` +
+            pull.body
+              ?.replace(REGENERATE_CHECKBOX_TEXT, '')
+              .replace(EMPTY_REGENERATE_CHECKBOX_TEXT, '')
+              .trim() ?? '',
+          WithRegenerateCheckbox.Yes
+        );
+        await octokit.pulls.update({
+          owner: destRepo.owner,
+          repo: destRepo.repo,
+          pull_number: pull.number,
+          title,
+          body,
+        });
+      } else {
+        // There's no open pull request; We'll need to create a pull request.
+        leftOvers.push(...yamlPathSet);
+      }
     }
-    return {kind: 'AppendedPullRequest', link: pulls.data[0].html_url};
-  } else {
-    return whatHappened;
+    yamlPathSets = [leftOvers];
+  } while (appendedPullCount > 0 && yamlPathSets[0].length > 0);
```

Comment:
I don't understand the condition for the `while` here. When is this condition met?
Is it worth adding a comment?